hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/router/RouterClientProtocol.java

```diff
@@ -1105,6 +1105,8 @@ private DatanodeStorageReport[] mergeDtanodeStorageReport(
             dnInfo.getLastUpdate() > oldDn.getDatanodeInfo().getLastUpdate()) {
           datanodesMap.put(nodeId, dn);
```

Comment:
If `dnInfo.getLastUpdate() > oldDn.getDatanodeInfo().getLastUpdate())`, should we still need to add `oldDn.getDatanodeInfo().getNumBlocks()` and `dnInfo.getNumBlocks()` ?